WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/lists/sections/granular/usecases/OverviewUseCase.kt

```diff
@@ -75,33 +77,38 @@ constructor(
 
     override fun buildStatefulUiModel(domainModel: VisitsAndViewsModel, uiState: UiState): List<BlockListItem> {
         val items = mutableListOf<BlockListItem>()
-        val selectedDate = uiState.selectedDate ?: domainModel.dates.lastOrNull()?.period
-        if (selectedDateProvider.getSelectedDate(statsGranularity) == null && selectedDate != null) {
-            selectedDateProvider.selectDate(
-                    statsDateFormatter.parseStatsDate(statsGranularity, selectedDate),
-                    statsGranularity
-            )
-        }
-        val selectedItem = domainModel.dates.find { it.period == uiState.selectedDate }
-                ?: domainModel.dates.lastOrNull()
-        items.add(
-                overviewMapper.buildTitle(
-                        selectedItem?.period,
-                        selectedDate,
-                        domainModel.period,
+        if (domainModel.dates.isNotEmpty()) {
+            val selectedDate = uiState.selectedDate ?: domainModel.dates.last().period
+            if (selectedDateProvider.getSelectedDate(statsGranularity) == null) {
+                selectedDateProvider.selectDate(
+                        statsDateFormatter.parseStatsDate(statsGranularity, selectedDate),
                         statsGranularity
                 )
-        )
-        items.add(
-                overviewMapper.buildChart(
-                        domainModel,
-                        statsGranularity,
-                        this::onBarSelected,
-                        uiState.selectedPosition,
-                        selectedDate
-                )
-        )
-        items.add(overviewMapper.buildColumns(selectedItem, this::onColumnSelected, uiState.selectedPosition))
+            }
+            val selectedItem = domainModel.dates.find { it.period == uiState.selectedDate }
+                    ?: domainModel.dates.last()
+            items.add(
+                    overviewMapper.buildTitle(
+                            selectedItem.period,
+                            selectedDate,
+                            domainModel.period,
+                            statsGranularity
+                    )
+            )
+            items.add(
+                    overviewMapper.buildChart(
+                            domainModel,
+                            statsGranularity,
+                            this::onBarSelected,
+                            uiState.selectedPosition,
+                            selectedDate
+                    )
+            )
+            items.add(overviewMapper.buildColumns(selectedItem, this::onColumnSelected, uiState.selectedPosition))
+        }
+        else {
+            AppLog.e(T.STATS, "There is no data to be shown in the overview block")
```

Comment:
I know this shouldn't happen anymore (because FluxC will emit an error and we'll never get invalid data) but I think we should be showing some kind of Error block in this case (or crash). If we don't, the whole page will miss the overview block and the other blocks will be loading indefinitely (because there is no selected date). We can also extend the selected date provider to emit something like "Invalid date" which would stop loading of the other blocks but I think that's totally out of scope for this PR.

Reply:
Good idea. Let's tackle this separately.